examples/demo/components/director/director.go

```diff
@@ -0,0 +1,150 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package director
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"time"
+
+	"open-match.dev/open-match/examples/demo/components"
+	"open-match.dev/open-match/internal/rpc"
+	"open-match.dev/open-match/pkg/pb"
+)
+
+func Run(ds *components.DemoShared) {
+	for !isContextDone(ds.Ctx) {
+		run(ds)
+	}
+}
+
+func isContextDone(ctx context.Context) bool {
+	select {
+	case <-ctx.Done():
+		return true
+	default:
+		return false
+	}
+}
+
+type status struct {
+	Status        string
+	LatestMatches []*pb.Match
+}
+
+func run(ds *components.DemoShared) {
+	defer func() {
+		r := recover()
+		if r != nil {
+			err, ok := r.(error)
+			if !ok {
+				err = fmt.Errorf("pkg: %v", r)
+			}
+
+			ds.Update(status{Status: fmt.Sprintf("Encountered error: %s", err.Error())})
+			time.Sleep(time.Second * 10)
+		}
+	}()
+
+	s := status{}
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Connecting to backend"
+	ds.Update(s)
+
+	conn, err := rpc.GRPCClientFromConfig(ds.Cfg, "api.backend")
+	if err != nil {
+		panic(err)
+	}
+	defer conn.Close()
+	be := pb.NewBackendClient(conn)
+
+	//////////////////////////////////////////////////////////////////////////////
+	s.Status = "Match Match: Sending Request"
+	ds.Update(s)
+
+	var matches []*pb.Match
+	{
+		req := &pb.FetchMatchesRequest{
+			Config: &pb.FunctionConfig{
+				Host: "om-demofunction",
+				Port: 50502,
+				Type: pb.FunctionConfig_GRPC,
+			},
+			Profile: []*pb.MatchProfile{
+				&pb.MatchProfile{
```

Comment:
This is enough. Run `make presubmit` to fix it.
```suggestion
				{
```